eng/testing/tests.mobile.targets

```diff
@@ -146,6 +157,32 @@
 
   </Target>
 
+  <Target Condition="'$(EnableAggressiveTrimming)' == 'true'"
+          Name="ConfigureTrimming"
+          BeforeTargets="PrepareForILLink">
+    <PropertyGroup>
+      <TrimMode>link</TrimMode>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <ManagedAssemblyToLink Include="$(OutDir)\*xunit*">
+        <IsTrimmable>true</IsTrimmable>
+        <TrimMode>link</TrimMode>
+      </ManagedAssemblyToLink>
+      <ManagedAssemblyToLink Condition="('%(ManagedAssemblyToLink.FileName).dll' != '$(MSBuildProjectName).dll')" >
+        <TrimMode>link</TrimMode>
+      </ManagedAssemblyToLink>
+      <ManagedAssemblyToLink Condition="('%(ManagedAssemblyToLink.FileName).dll' == '$(AssemblyName).dll')" >
+        <TrimMode>copy</TrimMode>
+      </ManagedAssemblyToLink>
+      <TrimmerRootAssembly Condition="'$(TargetOS)' == 'Android'" Include="AndroidTestRunner"/>
+      <TrimmerRootAssembly Condition="'$(TargetOS)' == 'Browser'" Include="WasmTestRunner"/>
+      <TrimmerRootAssembly Condition="'$(TargetOS)' == 'iOS'" Include="AppleTestRunner"/>
+      <TrimmerRootAssembly Include="$(AssemblyName)"/>
```

Comment:
These shouldn't be needed, I think, since the `PrepareForILLink` target will set these up for us.

Reply:
These are different settings. I think `PrepareForILLink` will set only `<TrimmerRootAssembly Include="$(AssemblyName)"/>`

Reply:
I was thinking that they might get picked up by https://github.com/dotnet/runtime/blob/32629120efd5beb6e174e312141f806e84720a2b/eng/testing/tests.mobile.targets#L222-L226 into `@(ResolvedFileToPublish)`

Reply:
I just tried removing L181 `<TrimmerRootAssembly Include="$(AssemblyName)"/>` and started seeing
```
 fail: 10:28:48.0231240 System.AggregateException: AggregateException_ctor_DefaultMessage ()

  fail: 10:28:48.0232360  ---> System.IO.FileNotFoundException:

  fail: 10:28:48.0232420 IO_FileName_Name, System.Buffers.Tests

  fail: 10:28:48.0232440    at System.Reflection.Assembly.Load(AssemblyName assemblyRef, StackCrawlMark& stackMark, AssemblyLoadContext assemblyLoadContext)

  fail: 10:28:48.0232480    at System.Reflection.Assembly.Load(AssemblyName assemblyRef)

  fail: 10:28:48.0232610    at Xunit.Sdk.ReflectionAssemblyInfo..ctor(String assemblyFileName)

  fail: 10:28:48.0232650    at System.Reflection.RuntimeConstructorInfo.InternalInvoke(Object obj, Object[] parameters, Boolean wrapExceptions)

  fail: 10:28:48.0232660 --- End of stack trace from previous location ---
  ```
  for `./dotnet.sh build /t:Test src/libraries/System.Buffers/tests/ /p:TargetOS=Browser /p:TargetArchitecture=wasm /p:Configuration=Debug /p:EnableAggressiveTrimming=true`

Reply:
Will address in a follow-up PR